QappTuning.py

```diff
@@ -0,0 +1,585 @@
+for f in os.listdir(os.path.expanduser("~")):
+    if picam2.global_camera_info()[0]["Model"] in f:
+        index_photo += 1
+if index_photo == 0:
+    folder_directory = os.path.join(os.path.expanduser("~"), picam2.global_camera_info()[0]["Model"])
+else:
+    folder_directory = os.path.join(os.path.expanduser("~"), picam2.global_camera_info()[0]["Model"] + "_" + str(index_photo))
+
+
+# what pi are we working on?
+if Picamera2.platform == Platform.PISP:
+    target = "pi5"
+    from ctt_pisp import json_template
+else:
+    target = "pi4"
+    from ctt_vc4 import json_template
+
+
+# checks if sting is a valid path to a folder
+def valid_path(s):
+    if (s[0] != "/" or s[-1] == "/"):
+        return False
+    else:
+        return True
+
+
+# create the screen that shades if it can se a macbeth plote clos to camera
+# macbeth reference data
+Cam = Camera("imx3.json", json=json_template)
+ref = cv2.imread(Cam.path + 'ctt_ref.pgm', flags=cv2.IMREAD_GRAYSCALE)
+ref_w = 120
+ref_h = 80
+rc1 = (0, 0)
+rc2 = (0, ref_h)
+rc3 = (ref_w, ref_h)
+rc4 = (ref_w, 0)
+ref_corns = np.array((rc1, rc2, rc3, rc4), np.float32)
+ref_data = (ref, ref_w, ref_h, ref_corns)
+
+# function that checks for macbeth close to camera
+
+
+def my_find_macbeth(img):
+    warnings.simplefilter("ignore")
+    fxn()
+    cor, mac, coords, msg = get_macbeth_chart(img, ref_data)
+    if cor < 0.75:
+        a = 2
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor < 0.75:
+        a = 4
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor > 0.5:
+        return (cor, coords)
+    else:
+        return (0, None)
+
+# future parameters
+
+
+pool = ThreadPoolExecutor(max_workers=4)
+overlay_active = False
+busy = False
+
+# Class that makes the wisget itself that will enclose macbeth if it sees it
+
+
+class MacbethWindow(QWidget):
+    done_signal = pyqtSignal()
+
+    def __init__(self, *args, **kwargs):
+        super(QWidget, self).__init__(*args, **kwargs)
+        self.qpicamera2 = QGlPicamera2(picam2, width=800, height=600, keep_ar=True, bg_colour=(236, 236, 236))
+        self.qpicamera2.done_signal.connect(self.capture_done)
+
+        self.window = QWidget()
+        self.layout_h = QHBoxLayout()
+        self.layout_h.addWidget(self.qpicamera2, 80)
+        self.window.resize(1200, 600)
+        self.window.setLayout(self.layout_h)
+
+        self._timer = QTimer()
+        self._timer.setInterval(20)
+        self._timer.timeout.connect(self.on_button_clicked)
+        self._timer.start()
+        picam2.start()
+
+        self.count = 0
+        self.futures = []
+        self.filter_machb = 0
```

Comment:
This name is a bit mysterious. Could we think of something better, and/or add a comment that would explain it?